examples/src/main/java/kafka/examples/ExactlyOnceMessageProcessor.java

```diff
@@ -215,6 +225,33 @@ private long getRemainingRecords(KafkaConsumer<Integer, String> consumer) {
         }).sum();
     }
 
+    private int retry(int retries, KafkaConsumer<Integer, String> consumer, ConsumerRecords<Integer, String> records) {
```

Comment:
Since we will not always retry. Maybe rename to `maybeRetry` ?

Reply:
Also, since this is an example for "general users", it'd be better we add some comments for this method above. Something like what this method is doing, why do we need this method, and what's the result exceeding max retry... etc.

Reply:
Agree.